# Enable shared library build
# Copyright (C) 2009-2016 Georgi D. Sotirov <gdsotirov@gmail.com>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
# Visit SlackPack at https://sotirov-bg.net/slackpack/
#
diff -urNad BLAS-3.6.0-orig/Makefile BLAS-3.6.0/Makefile
--- BLAS-3.6.0-orig/Makefile	2015-10-08 03:50:12.000000000 +0300
+++ BLAS-3.6.0/Makefile	2016-09-23 11:38:27.000000000 +0300
@@ -138,8 +138,7 @@
 	$(ZBLAS2) $(ZBLAS3) $(ALLBLAS)
 
 $(BLASLIB): $(ALLOBJ)
-	$(ARCH) $(ARCHFLAGS) $@ $(ALLOBJ)
-	$(RANLIB) $@
+	$(FORTRAN) -shared -Wl,-soname=$@ -o $@ $(ALLOBJ)
 
 single: $(SBLAS1) $(ALLBLAS) $(SBLAS2) $(SBLAS3)
 	$(ARCH) $(ARCHFLAGS) $(BLASLIB) $(SBLAS1) $(ALLBLAS) \
@@ -165,7 +164,7 @@
 	@FRC=$(FRC)
 
 clean:
-	rm -f *.o
+	rm -f *.o *.so.*
 
 .f.o: 
 	$(FORTRAN) $(OPTS) -c $< -o $@
diff -urNad BLAS-3.6.0-orig/make.inc BLAS-3.6.0/make.inc
--- BLAS-3.6.0-orig/make.inc	2011-04-19 22:08:00.000000000 +0300
+++ BLAS-3.6.0/make.inc	2016-09-23 11:38:27.000000000 +0300
@@ -16,7 +16,7 @@
 #  desired load options for your machine.
 #
 FORTRAN  = gfortran
-OPTS     = -O3
+OPTS     = $(FFLAGS) -fPIC
 DRVOPTS  = $(OPTS)
 NOOPT    =
 LOADER   = gfortran
@@ -31,4 +31,4 @@
 #
 #  The location and name of the Reference BLAS library.
 #
-BLASLIB      = blas$(PLAT).a
+BLASLIB      = libblas.so
